polyfill/lib/ecmascript.mjs

```diff
@@ -2347,6 +2348,272 @@ export const ES = ObjectAssign({}, ES2020, {
+      oneDayEarlier.millisecond,
+      oneDayEarlier.microsecond,
+      oneDayEarlier.nanosecond,
+      calendar,
+      largestUnit
+    ));
+    return ES.RoundDuration(
+      years,
+      months,
+      weeks,
+      days,
+      hours,
+      minutes,
+      seconds,
+      milliseconds,
+      microseconds,
+      nanoseconds,
+      1,
+      'days',
+      'ceil',
+      dtRelative
+    );
+  },
+  DifferenceZonedDateTime: (start, end, largestUnit, roundingIncrement, smallestUnit, roundingMode) => {
+    const ns1 = GetSlot(start, EPOCHNANOSECONDS);
+    const ns2 = GetSlot(end, EPOCHNANOSECONDS);
+    const nsDiff = ns2.subtract(ns1);
+    if (nsDiff.isZero()) return {};
+    const direction = nsDiff.divide(nsDiff.abs()).toJSNumber();
+
+    // Find the difference in dates only.
+    const timeZone = GetSlot(start, TIME_ZONE);
+    const calendar = GetSlot(start, CALENDAR);
+    const dtStart = ES.GetTemporalDateTimeFor(timeZone, GetSlot(start, INSTANT), calendar);
+    const dtEnd = ES.GetTemporalDateTimeFor(timeZone, GetSlot(end, INSTANT), calendar);
+    let { years, months, weeks, days } = ES.DifferenceDateTime(
+      GetSlot(dtStart, ISO_YEAR),
+      GetSlot(dtStart, ISO_MONTH),
+      GetSlot(dtStart, ISO_DAY),
+      GetSlot(dtStart, ISO_HOUR),
+      GetSlot(dtStart, ISO_MINUTE),
+      GetSlot(dtStart, ISO_SECOND),
+      GetSlot(dtStart, ISO_MILLISECOND),
+      GetSlot(dtStart, ISO_MICROSECOND),
+      GetSlot(dtStart, ISO_NANOSECOND),
+      GetSlot(dtEnd, ISO_YEAR),
+      GetSlot(dtEnd, ISO_MONTH),
+      GetSlot(dtEnd, ISO_DAY),
+      GetSlot(dtEnd, ISO_HOUR),
+      GetSlot(dtEnd, ISO_MINUTE),
+      GetSlot(dtEnd, ISO_SECOND),
+      GetSlot(dtEnd, ISO_MILLISECOND),
+      GetSlot(dtEnd, ISO_MICROSECOND),
+      GetSlot(dtEnd, ISO_NANOSECOND),
+      calendar,
+      largestUnit
+    );
+    let intermediateNs = ES.AddZonedDateTime(
+      GetSlot(start, INSTANT),
+      timeZone,
+      calendar,
+      years,
+      months,
+      weeks,
+      days,
+      0,
+      0,
+      0,
+      0,
+      0,
+      0,
+      'constrain'
+    ); // may disambiguate
+
+    // If clock time after addition was in the middle of a skipped period, the
+    // endpoint was disambiguated to a later clock time. So it's possible that
+    // the resulting disambiguated result is later than `this`. If so, then back
+    // up one day and try again. Repeat if necessary (some transitions are
+    // > 24 hours) until either there's zero days left or the date duration is
+    // back inside the period where it belongs. Note that this case only can
+    // happen for positive durations because the only direction that
+    // `disambiguation: 'compatible'` can change clock time is forwards.
+    while (
+      direction === 1 &&
+      ES.DurationSign(years, months, weeks, days, 0, 0, 0, 0, 0, 0) === 1 &&
+      intermediateNs.greater(ns2)
+    ) {
+      // TODO: after PlainDate.add rounding lands, uncomment use of relativeTo
```

Comment:
Is there an issue to track adding rounding to add/subtract methods?  Is there anything else remaining for rounding besides that?

Reply:
#856 is for that, that's why it's still open.